app/models/user.rb

```diff
@@ -9,4 +9,47 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friends, through: :friendships, foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    inverse_friends_array = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    (friends_array + inverse_friends_array).compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  end
+
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  end
+
+  def confirm_friend(user)
+    friendships = inverse_friendships.find { |friendship| friendship.user == user }
+    friendships.confirmed = true
+    friendships.save
+  end
```

Comment:
Writing it this way using ruby is good. But, in this milestone, we want you to use SQL query.  This will allow us to find user friends only by user_id in friendships table. Your code should be something like this:

has_many :confirmed_something, -> { where confirmed: true }, class_name: "Something"
has_many :things, through: :confirmed_something.